presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -206,6 +210,10 @@ public static ParquetPageSource createParquetPageSource(
             if (e instanceof PrestoException) {
                 throw (PrestoException) e;
             }
+            if (nullToEmpty(e.getMessage()).trim().equals("Filesystem closed") ||
```

Comment:
Will be good to know cases where we don't get FNFE but have to rely on exception string.

Reply:
Good point. But I am not quite sure if there are corner cases where we have to rely on exception string. I just move the exception handling out from the original code path.